### PR TITLE
Feature/registration each timestep

### PR DESCRIPTION
**Pull request by kjrush**
# PR Details
## Description

Implements behavior to send the registration message each timestep by using two sockets in parallel.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [x] I have added tests to cover my changes.
- [] All new and existing tests passed.

### CODE REVIEW

**Review by SaikrishnaBairamoni — DISMISSED** [review]

File: ns3_adapter/src/ns3_adapter.cpp [hunk 1]
```diff
@@ -77,7 +77,7 @@ void NS3Adapter::initialize() {
     ns3_client_.onMessageReceived.connect([this](std::vector<uint8_t> const &msg, uint16_t id) {onMessageReceivedHandler(msg, id); });
     
     // spin_rate = 50; TODO: temporaty change for test
```
Comment (saina-ramyar) [inline ns3_adapter/src/ns3_adapter.cpp]: we can remove the TODO now

File: ns3_adapter/src/ns3_adapter.cpp [hunk 2]
```diff
@@ -297,8 +291,13 @@ void NS3Adapter::pre_spin()
             connecting_ = false;
         }));
     }
-
-    
+    //if (!handshake_sent_)
```
Comment (saina-ramyar) [inline ns3_adapter/src/ns3_adapter.cpp]: The commented out code can be removed

File: ns3_adapter/src/ns3_adapter.cpp [hunk 3]
```diff
@@ -297,8 +291,13 @@ void NS3Adapter::pre_spin()
...
+    //{
+    ns3_reg_client_.connect(ns3_address_, ns3_registration_port_);
+    std::string handshake_msg = compose_handshake_msg(vehicle_id_, role_id_, std::to_string(ns3_broadcasting_port_), host_ip_);
```
Comment (saina-ramyar) [inline ns3_adapter/src/ns3_adapter.cpp]: the third input should have been "ns3_listening_port_".

**Review by saina-ramyar — CHANGES_REQUESTED** [review]
can you also change the "role_id_" to the correct format?

**Review by saina-ramyar — CHANGES_REQUESTED** [review]
can you also change the "role_id_" to the correct format?

**Comment by kjrush** [comment]
I think I've resolved all the comments if you want to take a second look for final approval

**Review by saina-ramyar — APPROVED** [review]

**Review by SaikrishnaBairamoni — APPROVED** [review]

**Pull request merged**